src/Symfony/Component/Yaml/Parser.php

```diff
@@ -359,6 +299,31 @@ public function parse($value, $flags = 0)
             mb_internal_encoding($mbEncoding);
         }
 
+        if (array_key_exists('<<', $data)) {
+            $merged = $data['<<'];
+            unset($data['<<']);
+
+            if (!is_array($merged)) {
+                throw new ParseException('YAML merge keys used with a scalar value instead of an array.');
+            }
+
+            // If the value associated with the merge key is a sequence, then this sequence is expected to contain mapping nodes
+            // and each of these nodes is merged in turn according to its order in the sequence. Keys in mapping nodes earlier
+            // in the sequence override keys specified in later mapping nodes.
+            if (isset($merged[0])) {
+                foreach ($merged as $mergedMapping) {
+                    if (!is_array($mergedMapping)) {
+                        throw new ParseException('Merge items must be arrays.');
+                    }
+                }
+                $merged = call_user_func_array('array_merge', array_reverse($merged));
+            }
+
+            // If the value associated with the key is a single mapping node, each of its key/value pairs is inserted into the
+            // current mapping, unless the key already exists in it.
+            $data = array_merge($merged, $data);
```

Comment:
I'm not sure that's a good idea because of numeric keys. Should we use the `+` operator?